indra/newview/llviewercontrollistener.cpp

```diff
@@ -158,7 +159,8 @@ void LLViewerControlListener::toggle(LLSD const & request)
 
 	if (info.control->isType(TYPE_BOOLEAN))
 	{
-		info.control->set(! info.control->get().asBoolean());
+        LL_WARNS("LLViewerControlListener") << "Toggling debug setting " << std::quoted(info.key) << LL_ENDL;
+		info.control->set(! info.control->get().asBoolean(), false);
```

Comment:
Let's capture `! info.control->get().asBoolean()` as a local variable first so we can report it in the `LL_WARNS()` before setting it.